src/main/java/org/apache/cassandra/sidecar/utils/SSTableImporter.java

```diff
@@ -310,6 +307,31 @@ private void cleanup(ImportOptions options)
                                                  cause));
     }
 
+    /**
+     * Aggregates pending imports per host from multiple keyspaces and tables
+     */
+    private void reportPendingImportAggregates()
+    {
+        Map<String, Integer> aggregates = new HashMap<>();
+        for (Map.Entry<ImportKey, ImportQueue> entry : importQueuePerHost.entrySet())
+        {
+            aggregates.compute(entry.getKey().host, (k, v) -> entry.getValue().size() + (v == null ? 0 : v));
+        }
+
+        aggregates.forEach((host, count) -> {
+            try
+            {
+                // Report aggregate metrics for the queues
+                InstanceMetadata instance = metadataFetcher.instance(host);
+                instance.metrics().sstableImport().pendingImports.metric.setValue(count);
+            }
+            catch (Exception e)
+            {
+                LOGGER.warn("Unable to report metrics for host={} pendingImports={}", host, count, e);
+            }
+        });
```

Comment:
also a nit
```suggestion
        importQueuePerHost
        .entrySet()
        .stream()
        .collect(Collectors.groupingBy(e -> e.getKey().host, Collectors.counting()))
        .forEach((host, count) -> {
            try
            {
                // Report aggregate metrics for the queues
                InstanceMetadata instance = metadataFetcher.instance(host);
                instance.metrics().sstableImport().pendingImports.metric.setValue(count.intValue());
            }
            catch (Exception e)
            {
                LOGGER.warn("Unable to report metrics for host={} pendingImports={}", host, count, e);
            }
        });
```

Reply:
I find this not as readable , so that's why I went the other way.